File: sample/jstro/benchmark/fib.js
// Classic recursive fib
function fib(n) {
  if (n < 2) return n;
  return fib(n - 1) + fib(n - 2);
}

var start = Date.now();
var n = 35;
var r = fib(n);
var elapsed = (Date.now() - start) / 1000;
console.log("fib(" + n + ") = " + r);
console.log("elapsed: " + elapsed + "s");

File: sample/astr/benchmark/fib.r
# Classic recursive fib benchmark.  Sized so the run completes in
# ~1 second on a modern x86_64 with the interpreter (and well under
# that with -c).
fib <- function(n) {
    if (n < 2) {
        n
    } else {
        fib(n - 1) + fib(n - 2)
    }
}
print(fib(36))
